composables/content-parse.ts

```diff
@@ -127,7 +129,29 @@ export function parseMastodonHTML(
   if (collapseMentionLink)
     transforms.push(transformCollapseMentions(status, inReplyToStatus))
 
-  return transformSync(parse(html), transforms)
+  const node = parse(html) as Node
+
+  if (cleanSharedLink) {
+    // filter out invisible spans
+    const filteredNode = node.children.filter((child: Node) => !!child.children)
+    const matchedIndex = lastChildLinkMatchesPreviewUrl(filteredNode, cleanSharedLink)
+
+    if (matchedIndex)
+      filteredNode[filteredNode.length - 1].children.splice(matchedIndex, 1)
+  }
```

Comment:
Could we use a transform here?

Reply:
I tried using a transform function for this but found it unnecessary because it goes through each child node. Whereas what I am going for only needs to check the very last child

Reply:
I agree that a transform may not be needed. I think we could merge and iterate. A possibility is to avoid removing the link here but instead add some markup so we can later hide it or show it in the UI.